packages/admin-ui/src/filters/filter.state.tsx

```diff
@@ -0,0 +1,120 @@
+import React, { useState, useCallback, useMemo, useRef, useEffect } from 'react'
+import { Item } from '@react-stately/collections'
+import { useListBox } from '@react-aria/listbox'
+import type { ListState } from '@react-stately/list'
+import { useListState } from '@react-stately/list'
+
+import type { PickerStateReturn } from '../picker'
+import { usePickerState } from '../picker'
+
+export function useFilterState<T extends FilterItem>(
+  props: UseFilterStateProps<T>
+): UseMultipleFilterReturn<T> {
```

Comment:
I think this is the right type
```suggestion
): UseFilterTypeReturn<T> {
```

Reply:
It's actually `UseMultipleFilterReturn` they have the same return. `UseFilterStateReturn` is actually a generic type extended for the 2 use cases (single and multiselect)